types/src/v0/impls/state.rs

```diff
@@ -688,30 +689,30 @@ impl HotShotState<SeqTypes> for ValidatedState {
             .resolve()
             .expect("Chain Config not found in validated state");
 
-        let Some(l1_finalized) = proposed_header.l1_finalized() else {
+        let Some(proposed_finalized) = proposed_header.l1_finalized() else {
             tracing::error!("L1 finalized has None value.");
             return Err(BlockError::InvalidBlockHeader);
         };
 
         let parent_finalized = parent_leaf.block_header().l1_finalized().unwrap().number();
-        if l1_finalized.number() <= parent_finalized {
+        if proposed_finalized.number() < parent_finalized {
             tracing::error!(
                 "L1 finalized not incrementing. parent: {}, proposal: {}",
                 parent_finalized,
-                l1_finalized.number()
+                proposed_finalized.number()
             );
         }
         // Wait for l1_finalized with a timeout.
-        if let Err(e) = future::timeout(Duration::from_secs(2), async {
-            instance
-                .l1_client
-                .wait_for_finalized_block(l1_finalized.number())
-                .await;
-        })
-        .await
-        {
-            tracing::error!("Wait for finalized, timeout: {}", e);
-            return Err(BlockError::InvalidBlockHeader);
+        let finalized = instance
+            .l1_client
+            .wait_for_finalized_block(proposed_finalized.number())
+            .await;
+
+        if finalized.hash() != proposed_finalized.hash() {
+            tracing::error!("Invalid proposal: l1_finalized hash mismatch");
+        }
+        if finalized.timestamp() != proposed_finalized.timestamp() {
+            tracing::error!("Invalid proposal: l1_finalized timestamp mismatch");
         }
```

Comment:
Nit: I think we can just do `if finalized != proposed_finalized`. Less error prone in case we ever add fields to the `L1BlockInfo` struct

Reply:
dc197d664cf7cc2368500b08ed584a52b4c9398f